File: pinyougou-interface/src/main/java/com/pinyougou/core/service/PayService.java
package com.pinyougou.core.service;

import cn.itcast.core.pojo.order.Order;

import java.util.Map;

public interface PayService {

    /**
     * 生成支付页面需要的二维码
     * @return
     */
    Map<String,String> createNative(String userName) throws Exception;

    /**
     * 生成用户支付订单页面需要的二维码
     * @return
     */
    Map<String,String> payOrder(String userName,Order order) throws Exception;

    /**
     * 查询订单
     * @param out_trade_no
     * @return
     */
    Map<String,String> queryPayStatus(String out_trade_no,String userName) throws Exception;
    /**
     * 关闭订单支付
     * @return
     */
    public Map<String,String> orderClose(String out_trade_no) throws Exception;
}

File: pinyougou-web-user/src/main/webapp/js/service/payService.js
app.service('payService',function($http){
	//本地支付
	this.createNative=function(orderId){
		return $http.get('pay/payOrder.do?orderId = ' + orderId);
	}
	
	//查询支付状态
	this.queryPayStatus=function(out_trade_no,orderId){
		return $http.get('pay/queryPayStatus.do?out_trade_no='+out_trade_no,orderId);
	}
});